scipy/spatial/tests/test_kdtree.py

```diff
@@ -217,46 +217,19 @@ def test_vectorized_query_multiple_neighbors(self):
         assert_(np.all(i[:, :, -s:] == self.kdtree.n))
 
     @pytest.mark.parametrize('r', [0.8, 1.1])
-    def test_single_query_all_neighbors(self, r):
+    def test_single_query_raises_for_k_none(self, r):
         np.random.seed(1234)
         point = np.random.rand(self.kdtree.m)
-        with pytest.warns(DeprecationWarning, match="k=None"):
+        with pytest.raises(ValueError, match="k must be an integer or a sequence of integers"):
```

Comment:
Ah, just saw the linter complain that those lines are too long. This will need to be fixed

Reply:
Probably easiest to just use a wildcard
```suggestion
        with pytest.raises(ValueError, match="k must be an integer or*"):
```

Reply:
FYI, the star is not even needed. Pytest matches any subset, not a strict regex.

Reply:
I know; I just find adding the asterisk shows intent more clearly, and matches the [docs](https://docs.pytest.org/en/latest/reference/reference.html#pytest.raises) better.

Reply:
But fine either way, as long as the linter is happy 😅